python/ray/tune/tune.py

```diff
@@ -477,6 +478,13 @@ class and registered trainables.
     set_verbosity(verbose)
 
     config = config or {}
+    try:
+        from ray.rllib.algorithms.algorithm_config import AlgorithmConfig
+    except ImportError:
+        AlgorithmConfig = type(None)
+    if config and isinstance(config, AlgorithmConfig):
+        config = config.to_dict()
+
```

Comment:
This is an expensive import that leads to the timeouts we see in CI. Can we remove this from this PR?

Reply:
I see, is it because importing anything from rllib will register all environments and rllib trainables?

I can remove it from the PR, but I'm not actually sure how to fix this. One possibility is:

```
if has_attr(config, "to_dict") and callable(config.to_dict):
    config.to_dict()
```

Reply:
Actually, I do need to include this in the PR or else all the rllib tests are failing. WDYT about the above workaround? @krfricke

Reply:
Update: We're going to maintain that Tune requires a dict input and fix the failing tests/examples to do the `to_dict` conversion.